src/assets/scripts/client/aircraft/FlightManagementSystem/LegModel.js

```diff
@@ -1,538 +1,780 @@
          * @private
          */
-        this.isProcedure = false;
+        this._procedureModel = null;
 
         /**
-         * Indicates the leg is for a holding pattern
+         * Array of `WaypointModel`s that have been passed (or skipped)
          *
-         * This property should only be used internally for easier
-         * switching on routeStrings.
+         * Aircraft will proceed along the route to each waypoint, and upon passing
+         * a waypoint, it will move that waypoint here to the `#_previousWaypointCollection`,
+         * and proceed to the next waypoint in the `#_waypointCollection` until no more
+         * `WaypointModel`s remain in the leg, at which point, they continue to the next leg.
          *
-         * The `WaypointModel` also has an `#isHold` property
-         * that should be used to determine if a waypoint is
-         * for a holding pattern.
-         *
-         * @property _isHold
-         * @type {boolean}
+         * @for LegModel
+         * @property _previousWaypointCollection
+         * @type {array<WaypointModel>}
+         * @default []
          * @private
          */
-        this._isHold = false;
-
-        /**
-         * When a leg is a procedure, this property describes what type of procedure
-         *
-         * @type {PROCEDURE_TYPE}
-         * @default ''
-         */
-        this.procedureType = '';
+        this._previousWaypointCollection = [];
 
         /**
-         * String representation of the current routeSegment.
-         *
-         * A directRoute contains a single WaypointModel and is separated
-         * by `..` in the routeString. In this example there are two directRoute
-         * segments, thus this routeString will result in two LegModels:
-         * - `FIXA..COWBY`
-         *
-         * A procedureRoute contains many `WaypointModel` objects and describes
-         * a standardRoute (sid/star/airway). procedureRoutes are separated by
-         * a single '.':
-         * - `DAG.KEPEC3.KLAS`
+         * Standard-formatted route string for this leg, excluding any special characters
          *
          * @for LegModel
-         * @property routeString
+         * @property _routeString
          * @type {string}
+         * @default ''
+         * @private
          */
-        this.routeString = '';
+        this._routeString = '';
 
         /**
-         * List of `WaypointModel` objects defined within a `LegModel`.
+         * Array of `WaypointModel`s to follow, excluding any waypoints passed (or skipped)
          *
-         * If this leg represents a `directRoute`, there will only be one
-         * `WaypointModel` contained in `#waypointCollection`.
+         * Upon completion of a given `WaypointModel` in the `#_waypointCollection`, that
+         * waypoint will be moved to the `#_previousWaypointCollection`, and the aircraft will
+         * continue to the next `WaypointModel` in the `#_waypointCollection`.
          *
```

Comment:
can we add a bit about the significance of the `0` index item in this collection?